### ISSUE TITLE
Extrinsics bytes value entered for setCode does not appear in signer modal when Submit Transaction

### ISSUE THREAD

**Issue by ltfschoen** [polkadot-js/apps#403]
Refer to below screenshot. A code value of `0x09` was entered, but the signer modal value is just `0x` 

<img width="1138" alt="screen shot 2018-11-05 at 18 42 42" src="https://user-images.githubusercontent.com/6226175/48016051-dd270c80-e12a-11e8-8363-859f93dbc96a.png">

Proposed to fix this based on experience in this PR https://github.com/polkadot-js/apps/pull/389

**Comment by jacogr** [comment]
Ok, so when you try and enter raw hex, you need to do so correctly. Effectively what you enter with `0x09` is the value 0x09 >> 2 (as length) with no bytes following.

So the API types decoding would take your value and try to make sense of it, the only sense it can make is the `0x` value.

With proper data/file, this happens -

<img width="922" alt="polkadot apps portal 2018-11-05 19-01-08" src="https://user-images.githubusercontent.com/1424473/48016982-57f12700-e12d-11e8-9f16-4e91029abbfa.png">

**Comment by ltfschoen** [comment]
Cool thanks, so it looks like we need input validation on the raw hex Bytes value (i.e. highlight the input field in 'red' and disable the "Submit Transaction" button until they enter/drag-and-drop/upload a valid value)?

**Comment by ltfschoen** [comment]
@jacogr When I to go http://localhost:3000/#/extrinsics, and create a transaction:
ALICE > councilVoting > propose(proposal) > consensus > setCode(new) > 0x43470d0a1a0a0000000d4948445200000734000002dc080600000081cb86ae00000c286943435049434320, 
and then click "Submit Transaction", I get error `Uncaught TypeError: Cannot read property 'toString' of null`, because 
`accountNonce` is `null` and on [this line of code in Extrinsic.tsx](https://github.com/polkadot-js/apps/blob/next/packages/ui-signer/src/Extrinsic.tsx#L43) we are calling `accountNonce.toString()`.

I am not sure where the value for `accountNonce` of `null` is set, but it appears to be passed down with <Signer> in polkadot-js/apps/packages/apps/src/index.tsx,
then in polkadot-js/apps/packages/ui-signer/src/index.tsx it's passed down within `queue` prop to <Modal>, which contains a queue of extrinsics of type `QueueTx`, and there we use `getDerivedStateFromProps` to assign it to `currentItem`, and in its `renderContent()` method its passed down to <ExtrinsicDisplay> packages/ui-signer/src/Extrinsic.tsx as prop `value`. 

Since the value of `accountNonce` is `null`, it's still considered a value (null, false and 0 are all considered values and do not trigger the ES6 destructuring default value to kick in), so it isn't assigned to the ES6 destructuring default value of `new BN(0)` [here](https://github.com/polkadot-js/apps/blob/next/packages/ui-signer/src/Extrinsic.tsx#L24). The ES6 destructuring default value of `new BN(0)` would only be assigned to `accountNonce` if `accountNonce` had a value of `undefined`.

As I'm not sure where `accountNonce` is initially set to a value of `null`, the work-around I've come up with so far is to do the following, which allows the "Submit Transaction" Modal to appear without error, however the hex code in the modal is `0x470...320` instead of `0x434...320`:
```
class Transaction extends React.PureComponent<Props> {
  render () {
    const { children, t, value: { accountId, accountNonce, extrinsic } } = this.props;

    ...

    let acctNonce = undefined;

    if (this.props.value && this.props.value.accountNonce === null) {
      acctNonce = new BN(0);
    } else {
      acctNonce = accountNonce;
    }

    const index = acctNonce && acctNonce.toString();

    ...

    return [
      <Modal.Header key='header'>
        ...
      </Modal.Header>,
      <Modal.Content className='ui--signer-Signer-Content' key='content'>
        ...
              <Trans i18nKey='decoded.short'>
                You are about to sign a message from ... with an index of <span className='code'>{index}</span>
              </Trans>
```

**Comment by jacogr** [comment]
@ltfschoen The latter is an additional issue, fixed here https://github.com/polkadot-js/apps/pull/407

**Comment by jacogr** [comment]
Reported issues closed in #405 & #407 (next branch)
